### PR TITLE
QE-181: Store provisioning services

### PR DESCRIPTION
**Pull request by avattipalli**
Jira: [QE-181](https://jira.bigcommerce.com/browse/QE-181)

## What? Why?
This change is to have **create store** service which makes a call to underlying create store service provided by bc-store-setup-node. 

## Additional Deploy Steps? (Optional)
None

## How was it tested?
Added new unit tests around the service

- [ ] Cloud Dev end-to-end - Describe Steps
- [x] Additional test cases (Optional)
- [ ] Integration - Describe Steps (Optional)

## Testing / Proof
...

## How can this be undone?
1. Revert this PR

ping @bigcommerce/billing-iam-testing

### CODE REVIEW

File: src/global-constants.js [hunk 1]
```diff
@@ -3,5 +3,7 @@ module.exports = {
   clientId: process.env.CLIENT_ID || 'id',
   clientSecret: process.env.CLIENT_SECRET || 'secret',
   defaultPassword: 'Password1!',
-  defaultTimeout: 10
+  defaultTimeout: 10,
+  STORE_CREATE_AUTH_TOKEN_ENDPOINT: '/api/v2/token',
```
Comment (aaron-humerickhouse) [inline src/global-constants.js:7]: be consistent with casing across the file

File: src/services/store-service.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        }
+    });
+}
+
+export async function createStore(account) {
+    try {
+        console.log(`\nCreating a store with the following account details: \n ${account}`);
+        const createStoreProvisioningPayload = _buildStoreProvisioningPayload(account);
+        const store = new StoreSetupService(HOST, clientId, clientSecret);
```
Comment (aaron-humerickhouse) [inline src/services/store-service.js:40]: misleading variable name

**Comment by aaron-humerickhouse** [comment]
Take a look at https://github.com/bigcommerce-labs/ng-marketplace/blob/master/tests/functional/services/provisioning-request.js

The provisioning takes a while so you have to poll to determine whether or not it's completed.

**Comment by avattipalli** [comment]
> https://github.com/bigcommerce-labs/ng-marketplace/blob/master/tests/functional/services/provisioning-request.js

Working on this update.

File: src/services/store-service.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            "sku": "STORE-LAUNCHBAY",
+            "account_id": account.account_id,
+            "options": {
+                "store": {
+                    "store_name": account.options.store.name,
+                    "email": account.email,
+                    "full_name": account.options.store.full_name,
+                    "country": account.options.store.country,
+                    "referral_id": "",
+                    "service_analytics_attributes": {
```
Comment (MarkJCram) [inline src/services/store-service.js:19]: Probably me not understanding the provisioning requirements, but us this node something that we'll eventually want to parameterize, or will we always just use static values?

**Comment by avattipalli** [comment]
Just noticed that I accidentally closed this PR, reopening it

File: src/services/store-service.js (hunk 3 above)
Comment (MarkJCram) [reply src/services/store-service.js:19]: Spoke with Anudeep and that's also what we landed on.  I don't necessarily like leaving static blocks of hard coded values, but I also couldn't find any tests that required any of these values to change.  Might revisit in the future, but for now it's fine

**Comment by avattipalli** [comment]
Working on fixing tests. Will request reviews once I get them fixed.

File: src/helpers/user-settings-helper.js [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
+import { DEFAULT_TIMEOUT } from '../global-constants';
 
 module.exports = function() {
 
     return actor({
 
         navigateToUserSettings() {
             this.click('Home');
-            this.waitForElement(selectors.my_profile_nav, defaultTimeout);
+            this.waitForElement(selectors.my_profile_nav, DEFAULT_TIMEOUT);
```
Comment (BC-Ubalderas) [inline src/helpers/user-settings-helper.js:10]: From reading the documentation, these WaitFor methods depend on the value defined on the codecept.conf.js file under the Webdriver options, so you could pass that value there instead, and then you don't have to define it everywhere. I might even say that unless we intend to use this timeout across other implicit waits, we probably don't need to add it as a separate constant outside of the conf file.

Comment (avattipalli) [reply src/helpers/user-settings-helper.js:10]: Yes, that is something as a group we can decide and is out of scope for this PR.

Comment (BC-Ubalderas) [reply src/helpers/user-settings-helper.js:10]: Sounds good!

Comment (MarkJCram) [reply src/helpers/user-settings-helper.js:10]: That was actually I was going to bring up at the meeting.  Having a default timeout defined there and used elsewhere is a good step, and even defining a separate constant within this file wouldn't be a bad step, either.  Either way, it's something we should define elsewhere and definitely outside the scope of the ticket

**Pull request merged**